src/UI/Components/Diagram/Context/ComposerEditorProvider.tsx

```diff
@@ -50,8 +50,14 @@ export const ComposerEditorProvider: React.FC<Props> = ({
 
   const serviceModels = useGetAllServiceModels(environment).useContinuous();
 
+  const mainModel = useMemo(
+    () => serviceModels.data?.find((service) => service.name === serviceName),
```

Comment:
We can't avoid that here, without wrapping it in some providers, which I would like to avoid, I feel like with `enabled` safeguard from react-query this will be more readable than having some logic of context in one component, and second part in other, nested component,